test/system/252-quadlet.bats

```diff
@@ -1803,19 +1803,28 @@ EOF
 
     # Table of drop-in .conf files. Format is:
     #
-    #    apply | dir | filename | [Section] | Content=...
+    #    apply | dir | parent dir (optional) | filename | [Section] | Content=...
+    #
+    # The parent dir used as regression test for https://github.com/containers/podman/issues/26555
     local dropin_files="
-y | toplevel  | 10 | [Unit]      | Description=Test File for Dropin Configuration
-n | toplevel  | 99 | [Install]   | WantedBy=default.target
-y | truncated | 50 | [Container] | ContainerName=truncated-dropins
-n | truncated | 99 | [Service]   | Restart=always
-n | truncated | 99 | [Install]   | WantedBy=multiuser.target
-y | quadlet   | 99 | [Service]   | RestartSec=60s
+y | toplevel  |        | 10 | [Unit]      | Description=Test File for Dropin Configuration
+n | toplevel  |        | 99 | [Install]   | WantedBy=default.target
+y | truncated |        | 50 | [Container] | ContainerName=truncated-dropins
+n | truncated |        | 99 | [Service]   | Restart=always
+n | truncated |        | 99 | [Install]   | WantedBy=multiuser.target
+y | quadlet   |        | 99 | [Service]   | RestartSec=60s
+n | toplevel  |        | 30 | [Service]   | Environment=test=wrong
+y | truncated | subdir | 30 | [Service]   | Environment=test=right
 "
 
     # Pass 1: Create all drop-in directories and files
-    while read apply dir file section content; do
-        local d="${quadlet_tmpdir}/${dropin_dirs[${dir}]}"
+    while read apply dir parent file section content; do
+        # By default parent will be '' when and not actually empty,
+        # make sure we set it to an empty dir instead.
+        if [[ "$parent" == "''" ]]; then
+            parent=""
+        fi
+        local d="${quadlet_tmpdir}/$parent/${dropin_dirs[${dir}]}"
```

Comment:
When `parent=""` this will lead to `local d="${quadlet_tmpdir}//${dropin_dirs[${dir}]}"`. Can the `//` lead to an issue?

Reply:
`//` and `/` it the exact same file for the kernel path resolving, I didn't want to make it more complicated with branching for this but I admit it can be a bit confusing.